samples/go/nomdex/vrange.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/attic-labs/noms/go/types"
+)
+
+type ventry struct {
+	v    types.Value
+	incl bool
+	inf  int8
+}
+
+type vrange struct {
+	lower ventry
+	upper ventry
+}
+
+type vrangeslice []vrange
+
+func (vs vrangeslice) Len() int {
+	return len(vs)
+}
+
+func (vs vrangeslice) Swap(i, j int) {
+	vs[i], vs[j] = vs[j], vs[i]
+}
+
+func (vs vrangeslice) Less(i, j int) bool {
+	return !vs[i].lower.equals(vs[j].lower) && vs[i].lower.isLessThanOrEqual(vs[j].lower)
+}
+
+func (vs vrangeslice) dbgPrint(w io.Writer) {
+	for i, r := range vs {
+		if i == 0 {
+			fmt.Fprintf(w, "\n#################\n")
+		}
+		fmt.Fprintf(w, "range %d: %s\n", i, r)
+	}
+	if len(vs) > 0 {
+		fmt.Fprintf(w, "\n")
+	}
+}
+func (r vrange) and(o vrange) (vslice vrangeslice) {
```

Comment:
Keep the method definitions with the structs. So define ventry/bound, then its methods, then vrange, then its methods, then vrangeslice, then it's methods.